### ISSUE TITLE
Going into baals room without delay after selling stuff in city? / varying messages to say to joining players?

### ISSUE THREAD

**Issue by loldermond** [kolton/d2bot-with-kolbot#2170]
1.
hi guys. i found this for baal to clean junk before visiting him - can someone do a similar thing for diablo with like a different portalnumber?

> if (Config.Baal.KillBaal) {
> 		Town.doChores();
> 		Town.move("portalspot");
> 		Pather.usePortal(131, null);
> 		if (Config.PublicMode) {
> 			say(Config.Baal.BaalMessage);
> 		}

2.
i got Config.DroppedItemsAnnounce.Enable = true and 5,7,8 in it. why doesn't he call out uniques dropping from monsters by name though? is it only for items you drop yourself?

3.
what symbols to put between messages for having a pool of different ones when someone for example gains a level in party?

thanks guys and girlguys

**Comment by Fa-b** [comment]
1.
```js
Pather.usePortal(108, null); // any blue Portal that leads to Chaos Sanctuary
```

2.
It's for items your bot drops after identifying them. What sense does it make for your leechers to know that there is some set ring (probably Cathan's Seal) where you killed mephisto before starting your Chaos run?

3. 
```
["Message 1", "Message 2", "Message 3"]; // comma
```

**Comment by loldermond** [comment]
if (Config.PublicMode) {

> 		say(Config.Diablo.DiabloMsg);
> 	}
> 
> 	this.diabloPrep();
> 	Attack.kill(243); // Diablo
> 	Pickit.pickItems();
> 
> 	Pather.teleport = Pather._teleport;
> 
> 	return true;

put this somewhere in here? like 

> if (Config.PublicMode) {
> 		say(Config.Diablo.DiabloMsg);
> 	}
> Town.doChores();
> Town.move("portalspot");
> Pather.usePortal(131, null);
> this.diabloPrep();
> 	Attack.kill(243); // Diablo
> 	Pickit.pickItems();
> 
> 	Pather.teleport = Pather._teleport;
> 
> 	return true;
> }

?

**Comment by loldermond** [comment]
1 more thing, why doesnt my bot say anything after baal?

// If LocalChat is enabled, chat can be sent via 'sendCopyData' instead of BNET
	// To allow 'say' to use BNET, use 'say("msg", true)', the 2nd parameter will force BNET
	// LocalChat messages will only be visible on clients running on the same PC
	Config.LocalChat.Enabled = false; // enable the LocalChat system
	Config.LocalChat.Toggle = false; // optional, set to KEY value to toggle through modes 0, 1, 2
	Config.LocalChat.Mode = 0; // 0 = disabled, 1 = chat from 'say' (recommended), 2 = all chat (for manual play)
	// If Config.Leader is set, the bot will only accept invites from leader. If Config.PublicMode is not 0, Baal and Diablo script will open Town Portals.
	Config.PublicMode = 1; // 1 = invite and accept, 2 = accept only, 3 = invite only, 0 = disable
	// Party message settings. Each setting represents an array of messages that will be randomly chosen.
	// $name, $level, $class and $killer are replaced by the player's name, level, class and killer
	Config.Greetings = ["Greetings $name,you look like a bag 0f fleas today!","Every beginning contains an ending, $name","Hey there $name, you radiate like a blushing sun today"]; // Example: ["Hello, $name (level $level $class)"]
	Config.DeathMessages = ["Watch out. Only daredevil of the day should be Diablo!","Die, Bart. Die","They killed $name. You bastards!","From drop that to drop dead"]; // Example: ["Watch out for that $killer, $name!"]
	Config.Congratulations = ["Congrats on your rising wisdom, $name.","Sometimes, a man has to get drunk to spend time with nimrods, $name - Hemmingway","Technics are a pinch to arrange the world the way we don't have to experience it, $class - Max Frisch","$level $name, The padawan will surmount his master."]; // Example: ["Congrats on level $level, $name!"]
	Config.ShitList = true; // Blacklist hostile players so they don't get invited to party.
	Config.UnpartyShitlisted = false; // Leave party if someone invited a blacklisted player.

**Comment by InterestingIndeed** [comment]
My guess: you need some brackets

`Config.LastMessage = ["Baal is a dead man"];`

**Comment by loldermond** [comment]
thanks. you also know how to set the text for using portal after dia kill?

**Comment by Fa-b** [comment]
Why don't you try as you suggested?

**Comment by InterestingIndeed** [comment]
When exactly do you want your char to sell stuff? Remind yourself that things that trigger Baal and Diablo are different (clearing throne vs activating seals). 

If you want your char to do town chores before Dia appears, i would put the "town chores" in before opening the last seals. This can be found in the baal.js script in line 467:

```
Attack.clear(30, 0, false, this.sort);
	this.vizierSeal();
	this.seisSeal();
	Precast.doPrecast(true);
	this.infectorSeal();
```

so it looks like this:

```
Attack.clear(30, 0, false, this.sort);
	this.vizierSeal();
	this.seisSeal();
**Town.doChores();
Town.move("portalspot");
Pather.usePortal(108, null);**
	Precast.doPrecast(true);
	this.infectorSeal();
```

**Comment by loldermond** [comment]
thanks to all. i wanted to do as suggested, just didnt know where to put it

**Comment by loldermond** [comment]
bot is pretty much saying just the first sentence. how can i randomize him using them all ("sentence1","sentence")?

ps where do i find the channel configuration to announce games?

**Comment by loldermond** [comment]
> 	if (Config.Baal.KillBaal) {
> 		Town.doChores();
> 		Town.move("portalspot");
> 		Pather.usePortal(131, null);
> 		if (Config.PublicMode) {
> 			say(Config.Baal.BaalMessage);
> 		}
> 
> 		Pather.moveTo(15090, 5008);
> 		**delay(5000);**
> 		Precast.doPrecast(true);
> 
> 		while (getUnit(1, 543)) {
> 			delay(500);
> 		}
> 

if i change (delay) to 3000, would that be 3 seconds? or how to make him go in faster after selling stuff?

**Comment by InterestingIndeed** [comment]
1. Dont change the topic every time a question is answered, that makes it hard for people who have the same question to find the post.
2. yes, that should make it shorter. 
3. you can always try what your code does, its a good way to learn.

**Comment by loldermond** [comment]
1. it still covers all the questions asked
2. thanks
3. okay

**Comment by Fa-b** [comment]
Usually the last few comments are the answer to the original question. if a question is answered, mark the issue as solved and/or close it. For a new question open a new issue.

**Comment by poisas** [comment]
> hi guys. i found this for baal to clean junk before visiting him - can someone do a similar thing for diablo with like a different portalnumber?
> 
> > if (Config.Baal.KillBaal) {
> > Town.doChores();
> > Town.move("portalspot");
> > Pather.usePortal(131, null);
> > if (Config.PublicMode) {
> > say(Config.Baal.BaalMessage);
> > }

thanks man, i was looking for this, issue is that default baal script is stupid, because if you run default ball.js it picks lots of magic rares from minions acording pickit and it goes to kill ball kills ball throws message in console trying to make room for "Shako" and quits game... lol..  and for exampe if there is some more special loot you want to pick how to force bot to pick them, or folowers to pick them before leader quits game ?
